src/main/java/uk/gov/pay/adminusers/service/SecondFactorAuthenticator.java

```diff
@@ -1,31 +1,55 @@
 package uk.gov.pay.adminusers.service;
 
+import com.google.inject.Inject;
 import com.warrenstrange.googleauth.GoogleAuthenticator;
 import org.slf4j.Logger;
 import uk.gov.pay.adminusers.logger.PayLoggerFactory;
 
+import java.time.Clock;
+
 import static com.google.common.io.BaseEncoding.base32;
 import static org.apache.commons.lang3.StringUtils.isBlank;
 
 
 public class SecondFactorAuthenticator {
 
     private static final Logger logger = PayLoggerFactory.getLogger(SecondFactorAuthenticator.class);
+    private static final int VALID_PAST_TIME_WINDOWS = 4; // Plus 1 and then times 30 for maximum validity
```

Comment:
Cool, I think variable name gives a good description of the purpose, I would suggest moving the `((VALID_PAST_TIME_WINDOWS + 1) * 30)` comment a bit nearer the logic that does the work?